core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -984,6 +987,20 @@ public boolean supportsAliasedValues() {
     return true;
   }
 
+  /**
+   * Returns whether the dialect needs cast in string operands of comparison operator. For instance,
+   * where employee_id = '10' is comparable in most of the dialect, so doesn't need cast for string
+   * operand '10'. But in BiqQuery the above statement is not valid without cast.
+   * @param node Operand of comparison operator which contain cast.
```

Comment:
contains ?

Reply:
Thank for pointing out, will update the description.